maat-court-data-api/src/main/java/gov/uk/courtdata/repository/OffenceRepository.java

```diff
@@ -23,21 +23,22 @@ public interface OffenceRepository extends JpaRepository<OffenceEntity, AsnSeqTx
     @Query(value = "SELECT COUNT(*) FROM MLA.XXMLA_OFFENCE WHERE CASE_ID = ?1 AND ASN_SEQ = ?2", nativeQuery = true)
     Integer getOffenceCountForAsnSeq(Integer caseId, String asnSeq);
 
-
-    @Query(value = "SELECT * FROM MLA.XXMLA_OFFENCE WHERE " +
-            "CASE_ID = :caseId AND OFFENCE_CODE = :offenceCode AND ASN_SEQ = :asnSeq " +
-            "AND TX_ID = (SELECT MAX( TX_ID ) FROM MLA.XXMLA_OFFENCE " +
-            "WHERE CASE_ID = :caseId AND OFFENCE_CODE = :offenceCode AND ASN_SEQ = :asnSeq)",
-            nativeQuery = true)
+    @Query(value =
+            "SELECT * FROM" +
+                    "(SELECT * FROM" +
+                    "(SELECT * FROM MLA.XXMLA_OFFENCE WHERE CASE_ID = :caseId AND ASN_SEQ = :asnSeq)" +
+                    "WHERE OFFENCE_CODE = :offenceCode ORDER BY TX_ID DESC)" +
+                    "WHERE ROWNUM < 2", nativeQuery = true)
     Optional<OffenceEntity> findByMaxTxId(Integer caseId, String offenceCode, String asnSeq);
 
     List<OffenceEntity> findByCaseId(Integer caseId);
 
-    @Query(value = "SELECT * FROM MLA.XXMLA_OFFENCE WHERE " +
-            "CASE_ID = :caseId AND OFFENCE_ID = :offenceId AND APPLICATION_FLAG = :applicationFlag " +
-            "AND TX_ID = (SELECT MAX( TX_ID ) FROM MLA.XXMLA_OFFENCE " +
-            "WHERE CASE_ID = :caseId AND OFFENCE_ID = :offenceId AND APPLICATION_FLAG = :applicationFlag)",
-            nativeQuery = true)
+    @Query(value =
+            "SELECT * FROM" +
```

Comment:
@Mhowell494  -  As discussed, please remove the nested/sub queries.

Reply:
@msahilan07 I've implemented the suggested change for both offence queries.